From c8d756a44c87f2453d272150d8c8ddaebcce5829 Mon Sep 17 00:00:00 2001
From: evilwombat <evilwombat@server.fake>
Date: Wed, 1 May 2013 00:20:23 -0700
Subject: [PATCH 05/13] ARM: Ambarella: Configure GPIOs for sdcard / buttons

Add the GPIO configuration needed to support the sdcard
slot and the power / record buttons on the Hero2 camera.

Signed-off-by: evilwombat <evilwombat@server.fake>
---
 arch/arm/mach-ambarella/init-generic.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/mach-ambarella/init-generic.c b/arch/arm/mach-ambarella/init-generic.c
index 2c3c929..5e95730 100755
--- a/arch/arm/mach-ambarella/init-generic.c
+++ b/arch/arm/mach-ambarella/init-generic.c
@@ -309,6 +309,16 @@ static void __init ambarella_init_generic(void)
 	}
 #endif
 
+	/* Needed for sdcard */
+	ambarella_gpio_config(83, GPIO_FUNC_SW_OUTPUT);
+	ambarella_gpio_set(83, 1);
+
+	/* Power button */
+	ambarella_gpio_config(48, GPIO_FUNC_SW_INPUT);
+
+	/* Record button */
+	ambarella_gpio_config(22, GPIO_FUNC_SW_INPUT);
+	ambarella_gpio_config(43, GPIO_FUNC_SW_INPUT);
 }
 
 /* ==========================================================================*/
-- 
1.7.9.5

